src/components/Explorer.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { browserHistory, Link } from 'react-router';
+import DocumentTitle from 'react-document-title';
+import Breadcrumbs from './Breadcrumbs';
+import Button from './Button';
+import InputSearch from './form/InputSearch';
+import { ADMIN_PREFIX } from '../constants';
+import { getDeleteMessage, getNotFoundMessage } from '../translations';
+
+export default class Explorer extends Component {
+  handleClickDelete(filename) {
+    const { deleteItem, params, type } = this.props;
+    const confirm = window.confirm(getDeleteMessage(filename));
+    const dir = params.splat ? `/${params.splat}` : '';
+    if (confirm) {
+      deleteItem(params.splat, filename);
+      browserHistory.push(`${ADMIN_PREFIX}/${type}${dir}`);
+    }
+  }
+
+  renderTitleCell(name, icon, type, path) {
+    return (
+      <td className="row-title">
+        <strong>
+          <Link to={`${ADMIN_PREFIX}/${type}/${path}`}>
+            <i className={`fa fa-${icon}`} aria-hidden="true" />
+            {name}
+          </Link>
+        </strong>
+      </td>
+    );
+  }
+
+  renderDirectoryRow(directory, type, index) {
+    const { name, relative_path } = directory;
+    return (
+      <tr key={index}>
+        {this.renderTitleCell(name, 'folder', type, relative_path)}
+        <td />
+      </tr>
+    );
+  }
+
+  renderFileRow(file, type, index) {
+    const { name, relative_path, http_url } = file;
+    return (
+      <tr key={index}>
+        {this.renderTitleCell(name, 'file-text-o', type, relative_path)}
+        <td>
+          <div className="row-actions">
+            <Button
+              onClick={() => this.handleClickDelete(name)}
+              type="delete"
+              icon="trash"
+              active={true}
+              thin
+            />
+            {http_url && (
+              <Button to={http_url} type="view" icon="eye" active={true} thin />
+            )}
+          </div>
+        </td>
+      </tr>
+    );
+  }
+
+  renderRows() {
+    const { items, type } = this.props;
+    return items.map((entry, index) =>
+      entry.type == 'directory'
+        ? this.renderDirectoryRow(entry, type, index)
+        : this.renderFileRow(entry, type, index)
+    );
+  }
+
+  render() {
+    const { items, type, newBtnLabel, search, params } = this.props;
+    const typeLabel = type === 'datafiles' ? 'data files' : type;
```

Comment:
We could create a constant for this later to get type labels;
```js
const TYPE_LABELS = {
  "datafiles": "data files",
  "pages": "pages",
  .. 
}
```

Reply:
Awesome idea!! :+1: